pennylane/devices/default_tensor.py

```diff
@@ -677,3 +714,84 @@ def execute_and_compute_vjp(
         raise NotImplementedError(
             "The computation of vector-Jacobian product has yet to be implemented for the default.tensor device."
         )
+
+
+@singledispatch
+def apply_operation_core(ops: Operation, device):
+    """Dispatcher for _apply_operation."""
+    device._quimb_mps.apply_gate(
+        qml.matrix(ops).astype(device._dtype), *ops.wires, parametrize=None
+    )
+
+
+@apply_operation_core.register
+def apply_operation_core_multirz(ops: qml.MultiRZ, device):
+    """Dispatcher for _apply_operation."""
+    apply_operation_core(qml.PauliRot(ops.parameters[0], "Z" * len(ops.wires), ops.wires), device)
+
+
+@apply_operation_core.register
+def apply_operation_core_paulirot(ops: qml.PauliRot, device):
+    """Dispatcher for _apply_operation."""
+    theta = ops.parameters[0]
+    wire_map = dict((w, i) for i, w in enumerate(ops.wires))
+    pw = qml.pauli.string_to_pauli_word(ops._hyperparameters["pauli_word"], wire_map)
+    cospsi = qml.math.cos(0.5 * theta) * copy.deepcopy(device._quimb_mps.psi)
+    for o in pw:
+        device._quimb_mps.apply_gate(
+            qml.matrix(o).astype(device._dtype), *o.wires, parametrize=None
+        )
+    device._quimb_mps._psi = cospsi - 1j * qml.math.sin(0.5 * theta) * device._quimb_mps.psi
```

Comment:
That is quite interesting, I didnt even think about just adding states together.
I see two problems with this though: 

1. This is effectively doubling the bond dimension (see [section 4.3 adding two MPS](https://arxiv.org/pdf/1008.3477))
<img width="653" alt="image" src="https://github.com/PennyLaneAI/pennylane/assets/43949391/9703ffeb-5e34-475f-8fd7-366fbdf862e4">
2. This is perhaps not compatible with full contraction, can you add states in full contraction mode? If yes, I could imagine its doing something inefficient like contracting with each branch of the sum, which means exponential branching for every paulirot gate added 😬 

Thats kind of the magic of MPOs that they can incorporate superposition of operators like that. I think we need to actually implement MPO decomposition of some operations in PennyLane, but this might be a bigger request perhaps beyond the scope of this PR

Reply:
1. Thanks for bringing this up, I forgot to add a call to `compress` after summing. That's what I had in mind to deal with this issue, but maybe there is a better way. 
2. The current PR is to resolve scalability issues in the MPS simulator only.

Reply:
yeah actually the more I think about it, the more I think for 2-term sums this might actually come down to more or less the same. For operators of the form $\sum_{i,j} P_i P_j$ it will make a bigger difference (because you can summarize it as low-bond-dim MPO whereas the sum may have many terms)